shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    disableCookiesUsage: true,
+                    cookieCfg: this._cookieMgrCfg
+                };
+                let manager = createCookieMgr(configValues);
+
+                let newKey = "test." + newId();
+                let newValue = newId();
+
+                // Set a cookie while disabled - should be cached
+                let result = manager.set(newKey, newValue);
+                Assert.equal(true, result, "Set should return true even when disabled (cached)");
+                Assert.equal(undefined, this._testCookies[newKey], "Cookie should not be in storage when disabled");
+
+                // Get should return the cached value
+                Assert.equal(newValue, manager.get(newKey), "Should return cached value when disabled");
+
+                // Enable cookies via dynamic config change - should flush cached values
+                configValues.disableCookiesUsage = false;
+
+                // Verify the cookie was flushed to actual storage
+                Assert.equal(newValue + "; path=/", this._testCookies[newKey], "Cookie should be flushed to actual storage via legacy config change");
+                Assert.equal(newValue, manager.get(newKey), "Should still return the value after enabling");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Multiple enable/disable cycles via config change work correctly",
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    cookieCfg: objExtend({}, this._cookieMgrCfg, {
+                        enabled: false
+                    })
+                };
+                let manager = createCookieMgr(configValues);
+
+                let newKey1 = "test1." + newId();
+                let newValue1 = newId();
+
+                // First cycle: disable -> cache -> enable -> flush
+                manager.set(newKey1, newValue1);
+                Assert.equal(newValue1, manager.get(newKey1), "Should return cached value");
+
+                configValues.cookieCfg.enabled = true;
+                Assert.equal(newValue1 + "; path=/", this._testCookies[newKey1], "First cookie should be flushed");
+
+                // Second cycle: disable -> cache -> enable -> flush
+                configValues.cookieCfg.enabled = false;
+                let newKey2 = "test2." + newId();
+                let newValue2 = newId();
+
+                manager.set(newKey2, newValue2);
+                Assert.equal(newValue2, manager.get(newKey2), "Should return second cached value");
+
+                configValues.cookieCfg.enabled = true;
+                Assert.equal(newValue2 + "; path=/", this._testCookies[newKey2], "Second cookie should be flushed");
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect